dali/python/nvidia/dali/external_source.py

```diff
@@ -153,8 +153,8 @@ class ExternalSource():
     again when the generator reaches end of iteration.
     In the case of the GPU input, it is the user responsibility to modify the
     provided GPU memory content only using provided stream (DALI schedules a copy on it
-    and all work is properly queued). If no stream is provided feeding input blocks until the
-    provided memory is copied to the internal buffer
+    and all work is properly queued). If no stream is provided, DALI will use a default, with
+    best-effort approach at correctness (see ``cuda_stream`` argument documentation for details).
```

Comment:
It's sort of orthogonal - we'll still block if using internal stream, because the kind of bug it protects against is even harder to detect. Maybe we should make it explicitly configurable. Let's discuss it on dev.

Reply:
I don't think we need that in python level. In C++ it is there. I would still mention here that if no stream is provided it will block.

Reply:
It's mentioned in the documentation of cuda_stream parameter.